lib/capybara/selector.rb

```diff
@@ -186,6 +194,7 @@ class Capybara::Selector; end # rubocop:disable Lint/EmptyClass
       add_error("Expected validation message to be #{msg.inspect} but was #{vm}") unless res
     end
   end
+  node_filter(:focused) { |node, value| (node.native == node.session.active_element.native) == value }
```

Comment:
Is there a way to detect or disabled a `node_filter` declaration when JavaScript is unavailable? This could either no-op, always return `false`, or `raise` and error from Rack Test. What options do we have?

Reply:
It should already raise an error on the calling of `active_element` if the user users the `focused` filter.  Also any reason you're calling `native` here?

Reply:
`node_filter(:focused, :boolean)...`

Reply:
> Also any reason you're calling native here?

Comparing the `node` and `node.session.active_element` is evaluating to false, since `node` is an instance of `Capybara::Node::Element`, and `node.session.active_element` evaluates to a Selenium-specialized instance.

Comparing the underlying native nodes evaluates to true.

Reply:
> It should already raise an error on the calling of active_element if the user users the focused filter. Also any reason you're calling native here?

From a developer's perspective, always raising makes a lot of sense. I'm concerned about CI and our test suite failing. Does this diff properly mark the tests to only execute for JS-enabled browsers?

Reply:
I've pushed up a change to add a `matches_focused_filter?` to `SelectorQuery`. Having a global filter was most appealing, but that means it's run on every query for every driver. Does a `SelectorQuery` instance have access to a mechanism to detect JavaScript? Would a `node_filter(:focused) { ... }` block?

Reply:
I fixed the code I merged around active_element into master to return an Element instance so the `.native` shouldn't be necessary anymore